test/Concurrency/actor_inout_isolation.swift

```diff
@@ -0,0 +1,105 @@
+// RUN: %target-typecheck-verify-swift -enable-experimental-concurrency
+// REQUIRES: concurrency
+
+// Verify that we don't allow actor-isolated state to be passed via inout
+// Check:
+//  - can't pass it into a normal async function
+//  - can't pass it into a first-class async function as a value
+//  - can't pass it into another actor method
+//  - can't pass it into a curried/partially applied function
+//  - can't pass it inout to a function that doesn't directly touch it
+//  - can't pass it into a function that was passed into the calling method
+//  - can't call async mutating function on actor isolated state
+
+struct Point {
+  var x: Int
+  var y: Int
+
+  mutating func setComponents(x: inout Int, y: inout Int) async {
+    defer { (x, y) = (self.x, self.y) }
+    (self.x, self.y) = (x, y)
+  }
+}
+
+actor class TestActor {
+  var position = Point(x: 0, y: 0)
+  var nextPosition = Point(x: 0, y: 1)
+  var value1: Int = 0
+  var value2: Int = 1
+}
+
+func modifyAsynchronously(_ foo: inout Int) async { foo += 1 }
+let modifyAsyncValue = modifyAsynchronously
+
+// external function call
+extension TestActor {
+
+  // Can't pass actor-isolated primitive into a function
+  func inoutAsyncFunctionCall() async {
+    // expected-error@+1{{actor-isolated property 'value1' cannot be passed 'inout' to asynchronous function}}
+    await modifyAsynchronously(&value1)
+  }
+
+  // Can't pass actor-isolated primitive into first-class function value
+  func inoutAsyncValueCall() async {
+    // expected-error@+1{{actor-isolated property 'value1' cannot be passed 'inout' to asynchronous function}}
+    await modifyAsyncValue(&value1)
```

Comment:
Might be worth adding another call-site into this method that tests a closure application, e.g., `{ (_ foo: inout Int) async in foo += 1 }(&value1)` to cover that part of the AST.

Reply:
added in the `inoutAsyncClosureCall` test case.